backend/src/DataSources/users-data-source.js

```diff
@@ -0,0 +1,65 @@
+const {DataSource} = require('apollo-datasource');
+const {createAccessToken} = require('../token')
+const crypto = require('crypto');
+const bcrypt = require('bcrypt');
+
+class User{
+  constructor(data){
+    this.id = crypto.randomBytes(16).toString('hex')
+    data.password = bcrypt.hashSync(data.password, 10);
+    Object.assign(this, data);
```

Comment:
Actually, it would be better to assign only white listed arguments here:
```suggestion
     const { id, name } = data
     Object.assign(this, { id, name, passwordHash });
```
that way you don't save the clear text password in the user.